source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -0,0 +1,286 @@
+#include "extensions/filters/http/jwt_authn/authenticator.h"
+
+#include "envoy/http/async_client.h"
+
+#include "common/common/logger.h"
+#include "common/http/message_impl.h"
+#include "common/http/utility.h"
+
+// from jwt_verify library
+#include "jwt.h"
+#include "verify.h"
+
+using ::google::jwt_verify::Status;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+// Extract host and path from a URI
+void extractUriHostPath(const std::string& uri, std::string* host, std::string* path) {
+  // Example:
+  // uri  = "https://example.com/certs"
+  // pos  :          ^
+  // pos1 :                     ^
+  // host = "example.com"
+  // path = "/certs"
+  auto pos = uri.find("://");
+  pos = pos == std::string::npos ? 0 : pos + 3; // Start position of host
```

Comment:
Is there an RFC for canonical URI parsing to reference?

Reply:
Added RFC